src/utils/spo.ts

```diff
@@ -694,5 +706,35 @@ export const spo = {
     const res = await request.get<{ AadObjectId: { NameId: string, NameIdIssuer: string } }>(requestOptions);
 
     return res.AadObjectId.NameId;
+  },
+
+  /**
+ * Retrieves a Custom Actions from a SharePoint site by Id.
+ * @param adminUrl URL of the SharePoint admin site
+ * @param camlQuery An optional viewQuery to add to the CAML query between the <Query> tags.
+ * @param viewFields An optional array of internal names of fields to include in the response.
+ */
+  getTenantSites(adminUrl: string, camlQuery?: string, viewFields?: string[]): Promise<TenantSites> {
+    let viewFieldsString: string = '';
+    viewFields?.forEach(v => {
+      viewFieldsString += `<FieldRef Name="${v}"/>`;
+    });
+
+    const query = `<View>${camlQuery}${viewFieldsString !== '' ? `<ViewFields>${viewFieldsString}</ViewFields>` : ''}</View>`;
```

Comment:
You don't need this variable assignment. You can move the entire statement to the request object.